pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -96,7 +100,30 @@ type RegistryPoll struct {
 	// Interval is used to determine the time interval between checks of the latest catalog source version.
 	// The catalog operator polls to see if a new version of the catalog source is available.
 	// If available, the latest image is pulled and gRPC traffic is directed to the latest catalog source.
-	Interval *metav1.Duration `json:"interval,omitempty"`
+	Interval     *metav1.Duration `json:"interval,omitempty"`
+	ParsingError string           `json:"-"`
+}
+
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface.
+func (u *UpdateStrategy) UnmarshalJSON(data []byte) (err error) {
+	var updateStrategyMap map[string]map[string]string
+	if err = json.Unmarshal(data, &updateStrategyMap); err != nil {
+		return err
+	}
+	registryPoll := &RegistryPoll{}
+	duration, err := time.ParseDuration(updateStrategyMap["registryPoll"]["interval"])
+	if err != nil {
+		registryPoll.ParsingError = fmt.Sprintf("error parsing spec.updateStrategy.registryPoll.interval. Setting the default value of %v. Error:%v", DefaultRegistryPollDuration, err)
+		defaultTime, parseErr := time.ParseDuration(DefaultRegistryPollDuration)
+		if parseErr != nil {
+			return parseErr
+		}
```

Comment:
Is it possible to parse the default a single time, say in `init()`, then deepcopy `defaultTime` for subsequent calls?

Also, failing to parse a default value is likely a non-recoverable programming error -- I'd probably `panic` in that case.

Reply:
`defaultTime` is only being parse once, here. There are no subsequent calls. 

I've changed the return statement to `panic` though.

Reply:
> defaultTime is only being parse once, here. There are no subsequent calls.

IIUC, it's called every time the custom marshaller is used, which is every time the resource is "read off the wire" (really constructed from the delta-fifo queue for most cases, but that's nitpicking)